docs/features/private-ip.md

```diff
@@ -1,30 +1,53 @@
 
-## Example
+There are two ways to configure the controller to use Private IP for ingress,
+
+## Assign to a particular ingress
+To expose a particular ingress over Private IP, use annotation [`use-private-ip`](../annotations.md#use-private-ip) in Ingress.
+
+### Usage
+```yaml
+appgw.ingress.kubernetes.io/use-private-ip: "true"
+```
+
+If Application Gateway doesn't have a private IP, then Ingresses with this annotation set to `true` will be pruned. This will be indicated in the ingress event and AGIC pod log.
+
+* Error as indicated in the Ingress Event
+
+    ```bash
+    Events:
+    Type     Reason       Age               From                                                                     Message
+    ----     ------       ----              ----                                                                     -------
+    Warning  NoPrivateIP  2m (x17 over 2m)  azure/application-gateway, prod-ingress-azure-5c9b6fcd4-bctcb  Ingress default/hello-world-ingress requires Application Gateway 
+    applicationgateway3026 has a private IP address
+    ```
+
+* Error as indicated in AGIC Logs
+
+    ```bash
+    E0730 18:57:37.914749       1 prune.go:65] Ingress default/hello-world-ingress requires Application Gateway applicationgateway3026 has a private IP address
+    ```
+
+
+## Assign Globally
+In case, requirement is to restrict all Ingresses to expose over Private IP, use `appgw.usePrivateIP: true` in `helm` config.
+
+### Usage
 ```yaml
 appgw:
     subscriptionId: <subscriptionId>
     resourceGroup: <resourceGroupName>
     name: <applicationGatewayName>
     usePrivateIP: true
-
-armAuth:
-     type: aadPodIdentity
-     identityResourceID: <identityResourceId>
-     identityClientID:  <identityClientId>
-
 ```
 
 This will make the ingress controller filter the ipconfigurations for a Private IP when configuring the frontend listeners on the Application Gateway.
-Controller will panic and crash if `usePrivateIP: true` and no Private IP is assigned.
+AGIC will panic and crash if `usePrivateIP: true` and no Private IP is assigned.
 
 **Notes:**
-
-1. Application Gateway v2 SKU manadates a Public IP. For meeting compliance requirement where the Application Gateway should be completely private, Attach a [`Network Security Group`](https://docs.microsoft.com/en-us/azure/virtual-network/security-overview) to the Application Gateway's subnet to restrict traffic.
-1. To expose the Ingress both on public and private, current recommedation is to deploy two Application Gateways with respective Ingress Controllers, one with Public IP and one with Private IP.
+Application Gateway v2 SKU manadates a Public IP. For meeting compliance requirement where the Application Gateway should be completely private, Attach a [`Network Security Group`](https://docs.microsoft.com/en-us/azure/virtual-network/security-overview) to the Application Gateway's subnet to restrict traffic.
```

Comment:
`compliance` could be one of the reasons.  I'd rephrase this as `Should you require Application Gateway to be private...`